### DISCUSSION TITLE
How to distribute my package/repo to others

### DISCUSSION THREAD

**Discussion by qqlearn123** [PyO3/pyo3#3126] [Questions]
I have been using `PyO3` for some time now. With this tool, the performance of my code improved a lot.

Up until now I am mainly using/testing it for myself locally, and I am currently at the stage of distributing my good shape package with others (a local, small working group).

So, I am wondering what are the steps should I follow to distribute my package/repo, such that what an end user needs to do is just `git clone` + `pip install` without worrying any `maturin/cargo` stuff?

Is it possible even if their machines/systems are potentially different from mine?

I do see there is an option to do source distribution with `maturin`, but still not quite sure what am I supposed to do after.

For a little more information, I am following the mixed project structure suggested in the maturin doc.

**Comment by davidhewitt** [comment]
The simplest option is to upload binary wheels to PyPI using maturin, which allows users with an up-to-date pip to install your package. You can use `maturin-action` in GitHub Actions to generate wheels for most machine/OS types.

If that isn't suitable, and you want to distribute source, it's inevitable that your users will need to install a Rust compiler (usually I'd advise doing this with rustup). If you set maturin as your build backend in your `pyproject.toml` then a `pip install` will automatically download a temporary install of `maturin` to orchestrate the build.
